### ISSUE TITLE
Lines are overwritten above the currently edited line when popup appears

### ISSUE THREAD

**Issue by pauluap** [davidhalter/jedi-vim#257]
Occasionally, when Jedi goes to autocomplete a element, it overwrites what seems like internal code onto the two lines above the line being edited.  The text being overwritten is

```
≡jedi=0, ≡                                  (***new_kw*) ≡jedi≡
≡jedi=0, ≡                                  (***new_kw*) ≡jedi≡
```

Where the first character is placed depends on the original lines, but if the original line is too long (80 chars minus the length of the evil text?), the original line is overwritten with the two lines above.

I used to think that it was random, but I noticed something that may be the trigger.  In the line being edited (shown in the screenshot below) there were two candidates for Jedi autocompletion, self.session.configure and self.engine.  I think that it works correctly when only one item needs autocompletion in a line.

![screenshot from 2014-02-28 21 08 16](https://f.cloud.github.com/assets/573966/2300673/b4218988-a100-11e3-956b-b6997089240c.png)

**Comment by davidhalter** [comment]
BTW as @bchretien noted, a workaround for this is to use call signatures in a different place, just set g:jedi#show_call_signatures to 2. By doing that they will appear in the status line.

**Comment by davidhalter** [comment]
Oh and I recently merged a change to the way how we insert those call signatures. Could you guys check if it works better now with the current jedi-vim master branch?

**Comment by graingert** [comment]
It now seems to be doing the same thing but with "== jedi ==" instead of tripple equals

![vim_jedi_strangeness](https://cloud.githubusercontent.com/assets/413772/6826725/64c0cd3a-d2fb-11e4-9dbf-e818b1e0b985.png)

**Comment by davidhalter** [comment]
Hehe, I realize that it's now buggy in general. Let me try to fix a few things and then we talk again. Sorry!

**Comment by graingert** [comment]
@davidhalter no wories, thanks for your great plugin!

**Comment by davidhalter** [comment]
Pull again. Better now?

**Comment by graingert** [comment]
Looks like it's fixed! :dancer:

**Comment by wliment** [comment]
@davidhalter  i  had update to the last version, but this problem remain exists.

look at the code at 68 line!  i have try another colorscheme  but no use to this.

![image](https://cloud.githubusercontent.com/assets/600490/6866314/ef1b8f94-d4b1-11e4-8960-df673e89251d.png)

![image](https://cloud.githubusercontent.com/assets/600490/6866278/951c96c8-d4b1-11e4-84cd-08673c963ff1.png)

**Comment by davidhalter** [comment]
colorschemes are not supported in Jedi. Can you try without one?

**Comment by davidhalter** [comment]
Also, can you give the exact reproduction case? What are you writing?

**Comment by graingert** [comment]
this feature seems to break undo/redo as well.

Is there a way to have dialogues without inserting text into the document?

**Comment by jhnphm** [comment]
I was able to reproduce this w/ a minimal vimrc and testcase w/ no syntax highlighting:

For the below testcase, inserting `filter(` in front of `match.group(2...` corrupts the line above, removing `tch(r'(\w+) ([:_\w]+) (.*)$`

    import re
    
    class Test:
        sef.operation_str = ""
        def __init__(self):
        def parse_opstring(self):
            match             = re.match(r'(\w+) ([:_\w]+) (.*)$', self.operation_str)
            self.macro_names  = match.group(2).split(':')
            self.prototypes   = match.group(3).split(':')




My testcase vimrc w/ Vundle:

    " set the runtime path to include Vundle and initialize
    set rtp+=~/.vim/bundle/Vundle.vim
    call vundle#begin()
    " " alternatively, pass a path where Vundle should install plugins
    " "call vundle#begin('~/some/path/here')
    "
    " " let Vundle manage Vundle, required
    Plugin 'gmarik/Vundle.vim'
    Plugin 'davidhalter/jedi-vim'
    
    " All of your Plugins must be added before the following line
    call vundle#end()            " required
    filetype plugin indent on    " required

python-jedi is 0.8.1 (latest in arch) , vim-jedi latest from git (d65c35a69d0fd7ae6e809e83a33b0d4eaf6b346f), using gvim7.4-659

**Comment by WoLpH** [comment]
I regularly get bitten by this bug, it corrupts files which makes it a really big issue but I seem to be unable to reliably reproduce it. It doesn't always happen but sometimes (when Jedi is a bit slow) I tend to get issues like these.

Anyone know of a way to prevent this from occurring completely? (or at least, reduce it to such a rarity that it doesn't matter too much anymore?). Is there no way to display anything in the editor without actually overwriting the line?

**Comment by davidhalter** [comment]
@graingert @WoLpH If you don't want history modifications, set `g:jedi#show_call_signatures = 2`. That makes the call signatures appear at the bottom of the window (but not above the current line, which I favor).

@jhnphm Sorry I cannot reproduce that :-(

**Comment by WoLpH** [comment]
Second that, I do prefer them to be above the current line. I've just updated my Vim and updated to the latest jedi-vim, perhaps it has some positive effect :)

**Comment by jhnphm** [comment]
@davidhalter Oddly enough, I can *not* reproduce it either running in a Ubuntu VM, only on my arch machine, with the same vimrc. Ubuntu is running patch 52 and arch is on 663.

**Comment by davidhalter** [comment]
@jhnphm So what you're saying is that with a newer version it's not working anymore even on **the same** system?

**Comment by jhnphm** [comment]
Hm, seems like probably exists between keyboard and chair. For whatever reason I assumed `:BundleInstall` would update jedi, but I needed to run `:BundleUpdate`

It seems for my testcase at least, the problem was fixed sometime between d65c35a69d0fd7ae6e809e83a33b0d4eaf6b346f and 43dcc4809f7c5280a55daa98ddd5c8415502d750

**Comment by jhnphm** [comment]
@davidhalter Hm, seems like probably exists between keyboard and chair. For whatever reason I assumed `:BundleInstall` would update jedi, but I needed to run `:BundleUpdate`, and I didn't clean out my ~/.vim/bundle when running w/ the minimal vimrc. That would explain why even a ubuntu binary and vim runtime copied over to Arch continued to break. 

It seems for my testcase at least, the problem was fixed sometime between d65c35a69d0fd7ae6e809e83a33b0d4eaf6b346f and 43dcc4809f7c5280a55daa98ddd5c8415502d750 , probably cc9837dc236872cd94303f140a3d9fcfdd9432bf

Thanks for making such a great piece of software :)
